e2e/actions/l2_challenger_test.go

```diff
@@ -182,6 +208,202 @@ interaction:
+	// If Proto Dank Sharding is introduced, the below code fix may be restored.
+	// block := proposer.SyncStatus().FinalizedL2
+	// outputOnL1, err := outputOracleContract.GetL2OutputAfter(nil, new(big.Int).SetUint64(block.Number))
+	targetBlockNum := big.NewInt(int64(testdata.TargetBlockNumber))
+	outputIndex, err := outputOracleContract.GetL2OutputIndexAfter(nil, targetBlockNum)
+	require.NoError(t, err)
+	outputOnL1, err := outputOracleContract.GetL2OutputAfter(nil, targetBlockNum)
+	require.NoError(t, err)
+	block, err := propEngine.EthClient().BlockByNumber(t.Ctx(), targetBlockNum)
+	require.NoError(t, err)
+	require.Less(t, block.Time(), outputOnL1.Timestamp.Uint64(), "output is registered with L1 timestamp of L2 tx output submission, past L2 block")
+	outputComputed, err := proposer.RollupClient().OutputAtBlock(t.Ctx(), targetBlockNum.Uint64())
+	require.NoError(t, err)
+	require.NotEqual(t, eth.Bytes32(outputOnL1.OutputRoot), outputComputed.OutputRoot, "output roots must different")
+
+	// deposit bond for challenger
+	challenger.ActDeposit(t, challengerInitialAmount)
+	includeL1Block(t, miner, challenger.address)
+
+	// check bond amount before create challenge
+	bond, err := valPoolContract.GetBond(nil, outputIndex)
+	require.NoError(t, err)
+	require.Equal(t, 0, big.NewInt(dp.DeployConfig.ValidatorPoolMinBondAmount.ToInt().Int64()).Cmp(bond.Amount))
+
+	// submit create challenge tx
+	txHash := challenger.ActCreateChallenge(t, outputIndex)
+
+	// include tx on L1
+	includeL1Block(t, miner, challenger.address)
+
+	// Check whether the submission was successful
+	receipt, err := miner.EthClient().TransactionReceipt(t.Ctx(), txHash)
+	require.NoError(t, err)
+	require.Equal(t, types.ReceiptStatusSuccessful, receipt.Status, "failed to create challenge")
+
+	challenge, err := colosseumContract.GetChallenge(nil, outputIndex)
+	require.NoError(t, err)
+	require.NotNil(t, challenge, "challenge not found")
+
+	// check bond amount doubled
+	bond, err = valPoolContract.GetBond(nil, outputIndex)
+	require.NoError(t, err)
+	require.Equal(t, 0, big.NewInt(2*dp.DeployConfig.ValidatorPoolMinBondAmount.ToInt().Int64()).Cmp(bond.Amount))
+
+interaction:
+	for {
+		status, err := colosseumContract.GetStatus(nil, outputIndex)
+		require.NoError(t, err)
+
+		switch status {
+		case chal.StatusChallengerTurn:
+			// do nothing to triggering challenger timeout
+			miner.ActEmptyBlock(t)
+		case chal.StatusAsserterTurn:
+			// call bisect by validator
+			txHash = validator.ActBisect(t, outputIndex)
+			includeL1Block(t, miner, validator.address)
+		case chal.StatusAsserterTimeout, chal.StatusReadyToProve:
+			// do nothing to triggering challenger timeout
+			miner.ActEmptyBlock(t)
+		case chal.StatusChallengerTimeout:
+			// check challenger bond amount decreased
+			cBal, err := valPoolContract.BalanceOf(nil, challenger.address)
+			require.NoError(t, err)
+			require.Equal(t, 0, new(big.Int).SetInt64(challengerInitialAmount-1).Cmp(cBal))
+			return
```

Comment:
If you return here, it will skip below check of the final status of challenge.
```suggestion
			break interaction
```